### PR TITLE
Automatically import icons from @material-ui/icons if package installed

### PR DESCRIPTION
**Pull request by dmtrKovalenko**
Get rid of required font-icon by default

### CODE REVIEW

**Comment by TrySound** [comment]
Shit, please don't. Never mix cjs with esm. Burn cjs everywhere. This will probably work only with bundlers which covers all possible and impossible hacks with interop.
Rollup will require some string/replace operations. I just got rid from similar shit in react-map-gl. Don't introduce the new one.

**Comment by dmtrKovalenko** [comment]
@TrySound then how we can achieve `optionalDependencies` work?

**Comment by TrySound** [comment]
material-ui icons is another large shit which cannot be converted to esm. I replaced it with a downloaded set of svg files and svgr transformer.

**Comment by TrySound** [comment]
Do not use optionalDependencies. material-ui core doesn't use it too.

**Comment by dmtrKovalenko** [comment]
But really a lot of our users are using the @material-ui/icons. And I want to get rid of icon fonts

**Comment by TrySound** [comment]
material-ui/icons are useless. svg files is the only necessary source of truth for icons IMO.

**Comment by TrySound** [comment]
Icons fonts in your project? Or in pickers?

**Comment by dmtrKovalenko** [comment]
But @material-ui/icons are also .svg file. They are svg components, no matter. 
But we live in the material-ui ecosystem...

Will dynamic import help?

**Comment by dmtrKovalenko** [comment]
I want to make an easy way for pickers to use svg icons automatically. But don't want to make a breaking change :)

**Comment by TrySound** [comment]
Nope. I'd say it's only for apps. Even core use internal icons.

**Comment by TrySound** [comment]
Do you mean via strings or what?

**Comment by TrySound** [comment]
I don't understand what is the problem now? You don't want to have own icons? They are just a few bytes.

**Comment by dmtrKovalenko** [comment]
Now we are rendering just `<Icon> icon_name </Icon>` by default. And if somebody wants to override the icon he needs to pass the icon component to the picker via props. 
That requires new wrapper to be created and passed down all props. That sucks. 
My idea was - when you already have `@material-ui/icons` installed icons will be imported from there.

**Comment by TrySound** [comment]
Let's talk in twitter DM

**Comment by dmtrKovalenko** [comment]
@TrySound looks better?

**Comment by TrySound** [comment]
Yeah, better. Just rename please PR

File: docs/src/Pages/Components/DateTimePicker/CustomDateTimePicker.example.jsx [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import React, { PureComponent, Fragment } from 'react';
 import { DateTimePicker } from 'material-ui-pickers';
-import { IconButton, Icon, InputAdornment } from '@material-ui/core';
-import { withUtilsService } from '../../../_shared/UtilsServiceContext';
+import { IconButton, InputAdornment } from '@material-ui/core';
```
Comment (TrySound) [inline docs/src/Pages/Components/DateTimePicker/CustomDateTimePicker.example.jsx]: Oopsy, path imports please. Or whole mui will appear in bundle

**Comment by dmtrKovalenko** [comment]
Ohhh, god, I did not save the rename 🤦‍♂️

File: docs/src/Pages/Components/DateTimePicker/CustomDateTimePicker.example.jsx (hunk 1 above)
Comment (dmtrKovalenko) [reply docs/src/Pages/Components/DateTimePicker/CustomDateTimePicker.example.jsx]: 🤦‍♂️

Comment (dmtrKovalenko) [reply docs/src/Pages/Components/DateTimePicker/CustomDateTimePicker.example.jsx]: But this is a docs file! @TrySound I am bundling full material-ui in docs 🤕

Comment (TrySound) [reply docs/src/Pages/Components/DateTimePicker/CustomDateTimePicker.example.jsx]: Okay then

**Pull request merged**